proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,532 @@
+
+        self.check_block_obj(m.fs.stream[0], m.stateblock_obj_dict)
+        self.check_block_obj_coverage(m.fs.stream[0], m.stateblock_obj_dict)
+
+    @pytest.mark.unit
+    def test_statistics(self, frame):
+        m = frame
+
+        assert number_variables(m) == m.statistics_dict['number_variables']
+        assert number_total_constraints(m) == m.statistics_dict['number_total_constraints']
+        assert number_unused_variables(m) == m.statistics_dict['number_unused_variables']
+        assert degrees_of_freedom(m) == m.statistics_dict['default_degrees_of_freedom']
+
+    @pytest.mark.unit
+    def test_general_methods(self, frame):
+        m = frame
+
+        # TODO: update for dynamic compatible property packages
+        # general methods requirements
+        required_general_methods_list = ['get_material_flow_terms',
+                                         'get_enthalpy_flow_terms']
+        tested_general_methods_list = []
+        for (method_name, ind) in m.general_methods_dict.keys():
+            if method_name not in tested_general_methods_list:
+                tested_general_methods_list.append(method_name)
+        for i in required_general_methods_list:
+            assert i in tested_general_methods_list
+
+        for (m_str, m_ind), (v_str, v_ind) in m.general_methods_dict.items():
+            assert hasattr(m.fs.stream[0], m_str)
+            general_method = getattr(m.fs.stream[0], m_str)
+            var = getattr(m.fs.stream[0], v_str)
+            if m_ind is None:
+                assert var[v_ind] == general_method(m_ind)
+            elif len(m_ind) == 1:
+                assert var[v_ind] == general_method(m_ind[0])
+            elif len(m_ind) == 2:
+                assert var[v_ind] == general_method(m_ind[0], m_ind[1])
+            else:
+                raise Exception(
+                    "general method {m_str} was assigned an index of {m_ind} with a length of "
+                    "{ind_length}, but a length of 2 or less is expected.".format(
+                        m_str=m_str, m_ind=m_ind, ind_length=len(m_ind)))
+
+    @pytest.mark.unit
+    def test_default_methods(self, frame):
+        m = frame
+
+        # default methods requirements
+        required_default_methods_list = ['default_material_balance_type',
+                                         'default_energy_balance_type',
+                                         'get_material_flow_basis']
+        for i in required_default_methods_list:
+            assert i in m.default_methods_dict
+
+        for k, v in m.default_methods_dict.items():
+            assert hasattr(m.fs.stream[0], k)
+            assert getattr(m.fs.stream[0], k)() == v
+
+    @pytest.mark.unit
+    def test_scaling(self, frame):
+        m = frame
+
+        calculate_scaling_factors(m.fs)
+
+        # check that all variables have scaling factors
+        unscaled_var_list = list(unscaled_variables_generator(m))
+        assert len(unscaled_var_list) == 0
```

Comment:
I just want to point out that the argument `include_fixed` is set to `False` by default. When testing the RO unit, I remember that setting `unscaled_variables_generator(m.fs.unit, include_fixed=True)` allowed us to catch certain variables that should've been scaled but weren't. That is, we expected the test to catch certain errors, and when it didn't; setting `include_fixed=True` resolved the issue.

However, doing this for the property models will pull up all of the parameters used as coefficients in empirical equations which do not have scaling factors.